includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -2390,6 +2390,11 @@ private function get_existing_compatible_payment_intent( $order, $payment_method
 			return null;
 		}
 
+		// Cash App Pay intents with a "requires payment method" status cannot be reused. See https://docs.stripe.com/payments/cash-app-pay/accept-a-payment?web-or-mobile=web&payments-ui-type=direct-api#failed-payments
```

Comment:
Would it be better to stop reusing payment intents altogether?

Payment intents seem inexpensive - we used to create a new one on each page load before deferred intents after all. 

We kept reusing them for deferred intents to maintain the behavior from before 8.0.0, but I can't think of any good reason to do it given the unexpected behaviors it brings, and all the logic we have in place for that. 

Last time I checked (about 3 months ago) WooPayments didn't reuse payment intents.

I can open a GH issue to address it separately if you agree.

Reply:
Yeah IDK. I'm not sure I understand why we added support for them to be reused originally. Looking at the function history, it's my understanding it was added for this issue: 

- https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2761. 

But I agree, if payment intents aren't a limited resource, it would be better to just create new ones rather than have this function which needs to keep up with the list of reasons why one can and cannot be reused.